projects/briey/memtestsim/src/main.c

```diff
@@ -0,0 +1,108 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <briey.h>
+
+#ifdef USE_BYTES
+    #define PATTERN   (0x55U)       // Repeating 0b0101...
+    #define TEST_T    uint8_t
+#else
+    #define PATTERN   (0x55555555U)
+    #define TEST_T    uint32_t
+#endif
+
+#define UART_DATA_LEN (8U)
+#define UART_BAUD     (115200U)
+
+extern void flushDataCache(); // From /vga/src/crt.S
+extern char _heap_start, _heap_end, _memTree_start, _memTree_end;
+volatile uint32_t state = 0;
+
+void fail(uint32_t errorAddr){
+    GPIO_A_BASE->OUTPUT = ~0x000000FF ^ 0x04000000; // All red LEDs (not LEDG[7:0], and toggle LEDG[8])
+    printf("Error detected at addr 0x%x\r\n", errorAddr);
+    while (1) {};
+}
+
+void pass(){
+    GPIO_A_BASE->OUTPUT = 0x000000FF; // All green LEDs
+    printf("MemTest PASS\r\n");
+    while (1) {};
+}
+
+void treePass(){
+    GPIO_A_BASE->OUTPUT = 0x0000000F; // Half of the green LEDs
+    printf("MemTest Tree PASS\r\n");
+}
+
+void main() {
+    GPIO_A_BASE->OUTPUT_ENABLE = 0xFFFFFFFF;
+    GPIO_A_BASE->OUTPUT = 0x04000000; // LEDG8
+
+    Uart_Config uartConfig;
+    uartConfig.dataLength = UART_DATA_LEN;
+    uartConfig.parity = NONE;
+    uartConfig.stop = ONE;
+    uartConfig.clockDivider = (CORE_HZ / UART_DATA_LEN / UART_BAUD) - 1;
+    uart_applyConfig(UART,&uartConfig);
+
+    register uint32_t currDataIdx = 0;
+    register TEST_T   testData;
+    register TEST_T   currPattern;
+
+    volatile TEST_T *heapStart = (TEST_T *)&_heap_start;
+    register TEST_T *heapEnd   = (TEST_T *)&_heap_end;
+    volatile TEST_T *treeStart = (TEST_T *)&_memTree_start;
+    register TEST_T *treeEnd   = (TEST_T *)&_memTree_end;
+
+    printf("heapStart: 0x%x\r\n", heapStart);
+    printf("heapEnd:   0x%x\r\n", heapEnd);
+    printf("treeStart: 0x%x\r\n", treeStart);
```

Comment:
same

Reply:
Going to keep these as such since it would still be good to know the tree location if we're explicitly not testing it.